components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -54,55 +62,175 @@ abstract class AbstractFetchDownloadService : CoroutineService() {
     @VisibleForTesting
     internal val context: Context get() = this
 
-    override fun onCreate() {
-        startForeground(
-            NotificationIds.getIdForTag(context, ONGOING_DOWNLOAD_NOTIFICATION_TAG),
-            DownloadNotification.createOngoingDownloadNotification(context)
-        )
-        super.onCreate()
+    private var listOfDownloadJobs = mutableMapOf<Long, DownloadJobState>()
+
+    // TODO: Can this be simplified/refactored?
+    data class DownloadJobState(
+        var job: Job? = null,
+        var state: DownloadState,
+        var currentBytesCopied: Long = 0,
+        var isPaused: Boolean = false,
```

Comment:
Maybe use an enum for ACTIVE/PAUSED/CANCELLED

Reply:
This is a great idea :)